### PR TITLE
Add InMemoryExporter for Metrics

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Add InMemoryExporter for Metrics
- Add an example for InMemoryExporter for Metrics

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: examples/Console/TestMetrics.cs [hunk 1]
```diff
@@ -70,6 +71,16 @@ internal static object Run(MetricsOptions options)
                         o.IsDelta = options.IsDelta;
                     });
             }
+            else if (options.UseExporter.ToLower() == "inmemory")
```
Comment (cijothomas) [inline examples/Console/TestMetrics.cs:74]: i'll probably leave this examples untouched. They are usually used for validating something visually (like in console or jaeger/zipkin etc).
InMemory is used for unittests typically. So we may use it in Unittests (which acts as usage example)

Comment (utpilla) [reply examples/Console/TestMetrics.cs:74]: I was also not sure if this was the right thing to do. I checked the existing `InMemoryExporter` example for reference and added it here.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/Console/TestInMemoryExporter.cs#L40

Comment (utpilla) [reply examples/Console/TestMetrics.cs]: I have removed the changes.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// </copyright>
+
+namespace OpenTelemetry.Exporter
+{
+    public class InMemoryExporterOptions
+    {
+        /// <summary>
+        /// Gets or sets the metric export interval.
+        /// </summary>
+        public int MetricExportInterval { get; set; } = 1000;
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.InMemory/InMemoryExporterOptions.cs:24]: not this PR, but we'll need to make the units obvious for the time interval! (similar to tracing. eg: ScheduledDelayMilliseconds)

**Review by cijothomas — APPROVED** [review]

**Pull request merged**